cram_pr2/cram_pr2_pick_place_demo/src/projection-demo.lisp

```diff
@@ -70,8 +76,37 @@
     (:spoon . ((-0.78 1.5 0.86) (0 0 0 1)))
     (:milk . ((-0.75 1.7 0.95) (0 0 0.7071 0.7071)))))
 
-
-(defun spawn-objects-on-sink-counter (&optional (spawning-poses *object-spawning-poses*))
+(defun make-poses-relative (&optional (spawning-poses *object-spawning-poses*))
+  (let* ((object-types '(:breakfast-cereal :cup :bowl :spoon :milk))
+         (map->surface (cl-transforms:pose->transform
+                       (btr:pose
+                        (btr:rigid-body
+                         (btr:object btr::*current-bullet-world* :kitchen) :|KITCHEN.sink_area_surface|))))
+         
+         (surface->objects (mapcar (lambda (type)
+                                     (setf coordinates (car (cdr (assoc type spawning-poses))))
+                                     (cl-transforms:make-transform
+                                      (cl-transforms:make-3d-vector
+                                       (first coordinates)
+                                       (second coordinates)
+                                       (third coordinates))
+                                      (cl-transforms:make-identity-rotation)))
+                                   object-types))
+         
+         (map->objects (mapcar (lambda (surface->object)
+                            (cl-transforms:transform* map->surface surface->object))
+                          surface->objects)))
+
+    (mapcar (lambda (transform type)
+              (setf tr (cl-transforms:translation transform))
```

Comment:
Same here, don't just put `setf` in a random place, where what you actually want is a `let` block.